app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -259,14 +257,19 @@ public function createOrder($reference, $session_quote_id = null)
 
         $quote->setTotalsCollectedFlag(false)->collectTotals()->save();
 
-        $existingOrder = Mage::getModel('sales/order')->loadByIncrementId($display_id);
+        /**
+         * TODO: Investigate if this is a problem.  Logically, it makes no sense that the order
+         *       exists at this point if we assume that the id is permanently reserved.  If the
+         *       id becomes available for use by another quote, this will be a problem.
+         */
+        $existingOrder = Mage::getModel('sales/order')->loadByIncrementId($order_id);
         if (sizeof($existingOrder->getData()) > 0) {
```

Comment:
I guess we should do something like
if (!empty($order->getData())) {

AS you do it in app/code/community/Bolt/Boltpay/controllers/ApiController.php::85

Reply:
Plugin should assume hook could be called multiple times. If HTTP request times out, Bolt server would assume it was failure and try again later. 
My assumption is this if block exists to handle 1) rare hook retries as described above and 2) race condition that leads to calling this method from both frontend and via hook.

Reply:
Re: `sizeof vs empty`, I slightly prefer `empty` for readability but IMO not a big deal.

Reply:
> Re: sizeof vs empty, I slightly prefer empty for readability but IMO not a big deal.

Me too.  But, let's take it to the next level and use the Magento method which is meant for this exact purpose:
`if (!$order->isEmpty())`

This will also be in the separate branch and pull request listed above.